### ISSUE TITLE
connection closed by peer on stopping the RTMP stream

### ISSUE THREAD

**Issue by talhatsk** [shogo4405/HaishinKit.swift#1472]
### Describe the bug

Hi, 
when i try to stop the stream,`FCUnpublish()` followed by `closeStream()` is invoked and the RTMP server reports `connection closed by peer` which is not the desired effect. It should stop/kill the stream. Then i tried it through OBS, when i stopped from there it sent `deleteStream() `instead of `closeStream()` after `FCUnpublish()` which deleted the stream and invoked Webhooks.
<img width="1025" alt="image" src="https://github.com/shogo4405/HaishinKit.swift/assets/118841446/da68cab4-1096-4281-8165-aa19e4bcf949">

<img width="1040" alt="Screenshot 2024-06-11 at 9 31 55 PM" src="https://github.com/shogo4405/HaishinKit.swift/assets/118841446/1ee56e56-3129-43b6-a427-ba35a36730d4">

### To Reproduce

stop the stream and check on wireshark, you will see `FCUnpublish()` followed by `closeStream()`

### Expected behavior

stopping the stream should call `deleteStream()` instead of `closeStream()`

### Version

1.8.1

### Smartphone info.

iPhone Xs Max

**Comment by shogo4405** [comment]
I see. I believe the fix is simple, so I'm looking forward to your PR.

**Comment by xoraingroup** [comment]
@shogo4405 - It seems it might need to rename the command to deleteStream() 

<img width="641" alt="image" src="https://github.com/shogo4405/HaishinKit.swift/assets/17129052/7e087590-dd7a-4fb5-a17a-13757b94baa9">


But I am not sure, if a separate method should be used to not break the API?

**Comment by xoraingroup** [comment]
@shogo4405 is this ok? I am waiting for a Yes or No :)

**Comment by shogo4405** [comment]
No. You need to consider the behavior when viewing, the behavior when using FCPublish, and the behavior when not using FCPublish. I have no knowledge regarding deleteStream, so I don't know. It will need to be investigated.

**Comment by xoraingroup** [comment]
closeStream closes stream but does not unpublish (delete) stream from rtmp server. This causes the rtmp server to not send webhooks because the stream is still there in rtmp server but in closed state and it waits for timeout of rtmp ingestion. The delete stream makes sure that the stream is unpublished from rtmp server and rtmp server sends the relevant webhooks to notify the relevant services. The webhooks are important part of creating flows and this is where we are stucked using this library.

From viewing perspective, the `FCUnpublish` with `deleteStream` unpublishes the stream and the client player just stops loading hls fragments. If fetching stream as rtmp pull, it just stops with error that stream does not exist.

I am not sure if i am making sense.

**Comment by levs42** [comment]
It makes sense in the context of your specific RTMP server. There are many of them, and they tend to have different implementations of RTMP spec. Therefore the best options is to create a PR with an optional `deleteStream` call. Here #1393 is a good example of handing undocumented `FCPublish` command in an optional manner.
